integration/kubernetes/confidential/agent_image.bats

```diff
@@ -181,6 +181,15 @@ assert_logs_contain() {
 	assert_logs_contain 'Validate image failed: \[PublicKeyVerifier { key: CosignVerificationKey'
 }
 
+# This test should be run first before the following 2 tests
```

Comment:
I don't like relying on the order of tests for them to pass as it is fragile. I thought that the setup run between tests should have reset the state anyway? If not we need to look at fixing this.

Reply:
Yeah, as far as I figured it out, [setup()](https://github.com/kata-containers/tests/blob/f07f559f7ea93798458865bbeeca8173e00f6f47/integration/kubernetes/confidential/agent_image.bats#L75-L94) resets a configuration file, but a root file system for an image/initrd.  In [cp_to_guest_img()](https://github.com/kata-containers/tests/blob/f07f559f7ea93798458865bbeeca8173e00f6f47/lib/common.bash#L274-L343), the image/initrd is modified and used for test, but it is not restored during the setup(), which leads to dependency between the tests. I am not sure if this was intentional. But you are right. We could fix this in a separate PR first.

Reply:
I guess that leads me to the question of how they worked before you moved them? Is the difference due to the agent-config.toml now being added in the rootfs.sh rather than us doing that during the test time?

Reply:
Interestingly I've just spotted that the crictl test have the follow in their teardown:
https://github.com/kata-containers/tests/blob/f07f559f7ea93798458865bbeeca8173e00f6f47/integration/containerd/confidential/agent_api.bats#L79-L81
so maybe we can just add this to the Kubernetes teardown as well to resolve the problem.
@wainersm - can you recall if there is a difference in the way that the guest image setup works for crictl and Kubernetes tests?